File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;

/**
 * 品牌服务层接口
 * 
 * @author Reasonless
 *
 */
public interface BrandService {


	/**
	 * 品牌添加
	 * 
	 * @param tbBrand
	 */
	void addBrand(TbBrand tbBrand);

	/**
	 * 根据id查询对应的数据
	 */
	TbBrand findOne(Long id);

	/**
	 * 品牌信息的更新
	 * 
	 * @param tbBrand
	 */
	void updateTbBrand(TbBrand tbBrand);
	
	/**
	 * 品牌删除
	 * @param ids
	 */
	void deleteTbBrand(Long[] ids);
	
	/**
	 * 根据条件查询，以及页面加载的时候分页
	 * @param tbBrand
	 * @param page
	 * @param pageSize
	 * @return
	 */
	PageResult findPage(TbBrand tbBrand,Integer page, Integer pageSize);
	
	/**
	 * 返回下拉列表
	 * @return
	 */
	List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//为了达到代码的复用和维护性，将服务进行抽取
app.service("brandService", function($http) {
	this.search = function(page, size, searchEntity) {
		return $http.post('../brand/search.do?page=' + page + '&size=' + size,
				searchEntity);
	}
	this.add = function(entity) {
		return $http.post('../brand/add.do', entity);
	}
	this.update = function(entity) {
		return $http.post('../brand/update.do', entity);
	}
	this.findOne = function(id) {
		return $http.get("../brand/findOne.do?id=" + id);
	}
	this.del = function(selectIds) {
		return $http.get('../brand/delete.do?ids=' + selectIds);
	}
	this.selectOptionList = function() {
		return $http.get('../brand/selectOptionList.do');
	}
});